src/components/Accordion/AccordionIcon.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import cn from 'classnames';
+
+import classes from './AccordionIcon.module.css';
+import { useAccordionContext, AccordionIconVariant } from './Context';
+import { ReactComponent as Arrow } from './expand-collapse.svg';
+import { ReactComponent as CircleArrow } from './CircleArrow.svg';
+
+export interface AccordionIconProps {
+  height?: string;
+  width?: string;
+}
+
+export const AccordionIcon = ({
+  height = '20',
+  width = '20',
+}: AccordionIconProps) => {
+  const { onClick, open, iconVariant } = useAccordionContext();
+  const props = { height, width };
+  const iconClassnames = [
+    classes[`accordion-icon`],
+    {
+      [classes['accordion-icon--opened']]: open,
+    },
+  ];
+  switch (iconVariant) {
+    case AccordionIconVariant.Primary:
+      return (
+        <Arrow
+          className={cn(iconClassnames)}
+          {...props}
+          onClick={onClick}
+        />
+      );
+    case AccordionIconVariant.Secondary:
+      return (
+        <CircleArrow
+          className={cn(iconClassnames)}
+          {...props}
+          onClick={onClick}
+        />
+      );
+  }
+};
```

Comment:
Why not spread `onClick` and `className` as well?

```suggestion
  const { onClick, open, iconVariant } = useAccordionContext();
  const iconClassnames = [
    classes[`accordion-icon`],
    {
      [classes['accordion-icon--opened']]: open,
    },
  ];
  const props = { 
  	height: 20, 
  	width: 20, 
		className: cn(iconClassnames),
		onClick,
	};
  
  switch (iconVariant) {
    case AccordionIconVariant.Primary:
      return (
        <Arrow
          {...props}
        />
      );
    case AccordionIconVariant.Secondary:
      return (
        <CircleArrow
          {...props}
        />
      );
  }
};
```